adapters/httpapi_winhttp.c

```diff
@@ -128,26 +129,37 @@ HTTPAPI_RESULT HTTPAPI_Init(void)
 
     if (nUsersOfHTTPAPI == 0)
     {
-        if ((g_SessionHandle = WinHttpOpen(
-            NULL,
-#if defined _MSC_VER && _MSC_VER < 1600
-            WINHTTP_ACCESS_TYPE_DEFAULT_PROXY,
-#else
-            WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY,
+        g_SessionHandle = NULL;
+#ifdef WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY
+        g_SessionHandle = WinHttpOpen(
+          NULL,
+          WINHTTP_ACCESS_TYPE_AUTOMATIC_PROXY,
+          WINHTTP_NO_PROXY_NAME,
+          WINHTTP_NO_PROXY_BYPASS,
+          0);
 #endif
-            WINHTTP_NO_PROXY_NAME,
-            WINHTTP_NO_PROXY_BYPASS,
-            0)) == NULL)
+        if (g_SessionHandle == NULL)
         {
-            LogErrorWinHTTPWithGetLastErrorAsString("WinHttpOpen failed.");
-            result = HTTPAPI_INIT_FAILED;
+	    g_ProxyAutomatic = 0;
+            /*Automatic option is not supported on Windows 7 or earlier so try older option*/
+            g_SessionHandle = WinHttpOpen(
```

Comment:
Indeed please move this whole block to within the #else

Reply:
If this was in the #else, that would cause a problem.  I want this code to be executed even if this code is compiled on a system where AUTOMATIC is available because it might be ran on a system (like Win 7) where automatic is not available.  It's confusing as I'm trying to allow for both the system it's compiled on and also the system it runs on.  Perhaps I need to do a more major change to make it clearer?